Automattic/woocommerce-payments
# Multicurrency: Do not modify deposit percentage for display.

Fixes: WooCommerce Deposits #\506

#### Changes proposed in this Pull Request







As reported in  WooCommerce Deposits #\506, enabling multi-currency modifies the display of percentage deposits.

The issue comes from `WooCommerceDeposits::modify_cart_item_deposit_amount_meta()` attempting to convert a percentage number although it were currency.

A store with a USD base selling a product with a 20% deposit results in the following displaying for select currencies:

AUD: 31%
CAD: 27%
INR: 1656%

I haven't written any tests and the new change relies on detecting a backtrace to the Deposits extension. 





**Before**
<img width="667" alt="Screen Shot 2023-08-04 at 2 58 01 pm" src="https://github.com/Automattic/woocommerce-payments/assets/519727/b062b4e6-4676-40f5-8600-2f666b739a85">

**After**
<img width="699" alt="Screen Shot 2023-08-04 at 2 58 52 pm" src="https://github.com/Automattic/woocommerce-payments/assets/519727/0fe15528-e2cc-450f-8c2e-3ae6f4998680">




#### Testing instructions



1. Configure Multi-currency Payments: using USD (default) and INR will best demonstrate the bug
2. Enable the WooCommerce Deposits extension
3. Configure and publish product to accept deposits:
    * Enable Deposits: Yes - Deposits are optional
    * Deposit Type:: percentage
    * Deposit Amount: 20
    * Default Deposit Selected Type: Pay Deposit
4. View the product on the front end
5. Switch to USD if required, observe the deposit amount is displayed as 20%
6. Switch the currency from USD to INR
7. Observe the deposit amount is displayed as significantly over 100% (the exact amount will vary according to currency fluctuations)



*

-------------------


**Post merge**
